data/battle_scripts_1.s

```diff
@@ -3078,23 +3118,8 @@ BattleScript_EffectHitEscape:
 	moveendfrom MOVEEND_TARGET_VISIBLE
 	jumpifbattleend BattleScript_HitEscapeEnd
 	jumpifbyte CMP_NOT_EQUAL gBattleOutcome 0, BattleScript_HitEscapeEnd
-	jumpifbattletype BATTLE_TYPE_ARENA, BattleScript_HitEscapeEnd
```

Comment:
Notice how here the battle script was jumping to `BattleScript_HitEscapeEnd`, but in your new `BattleScript_MoveSwitch` it's jumping to `BattleScript_MoveEnd`. This is fundamentally different because `BattleScript_HitEscapeEnd` does not call the moveend effects again while `BattleScript_MoveEnd` does.

Reply:
Yes! Thank you! Still trying to get a grip on the macros and these scripts. I updated `BattleScript_MoveSwitch` to have its own `-End` instruction that removed the `moveend` effects. Hopefully that is progress.